worker/sshserver/manifold.go

```diff
@@ -88,13 +98,18 @@ func (config ManifoldConfig) startWrapperWorker(context dependency.Context) (wor
 	}
 
 	w, err := config.NewServerWrapperWorker(ServerWrapperWorkerConfig{
-		NewServerWorker: config.NewServerWorker,
-		Logger:          config.Logger,
-		FacadeClient:    client,
+		NewServerWorker:      config.NewServerWorker,
+		Logger:               config.Logger,
+		FacadeClient:         client,
+		NewSSHServerListener: config.NewSSHServerListener,
 	})
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
-
 	return w, nil
 }
+
+// NewSSHServerListener returns a listener based on the given listener.
+func NewSSHServerListener(l net.Listener, t time.Duration) net.Listener {
```

Comment:
I don't understand this? Also t isn't used?

Reply:
See: https://github.com/juju/juju/pull/19118/files#diff-7eee8d6808d3d5cd339200cd303009cbaf050fcbf8b0790689c48a36d690d98aR793

Reply:
Ah I see, this signature is just for the testing listener and timeout is purely for context cancelling.